src/common/store-update-message-hub.ts

```diff
@@ -22,15 +22,16 @@ export class StoreUpdateMessageHub {
         this.registeredUpdateListeners[storeId] = listener;
     }
 
-    public readonly handleMessage = (message: StoreUpdateMessage<any>): void => {
+    public readonly handleMessage = (message: StoreUpdateMessage<any>): Promise<void> => {
```

Comment:
It feels odd to me to artificially return a promise from this method when it isn't always a top-level message listener (ex. in DevToolMessageDistributor). Could this function stay synchronous and be wrapped in an async function instead when it's the only listener?

Reply:
Yes, I think for our purposes it will be important to expand the return types of our assorted intermediate message handlers (eg, this and `Interpreter`) to be able to distinguish between 3 cases:

* "I didn't handle this message"
* "I handled this message via an async handler; here is the `Promise` tracking its work"
* "I handled this message via a fire-and-forget handler"

This is essentially what `PostMessageContentHandler` already does by returning a `BackchannelMessageResponse`; I would support generalizing and re-using it.